src/gitHubContributorFetcher.ts

```diff
@@ -0,0 +1,102 @@
+import { Octokit } from '@octokit/rest';
+
+export interface GitHubContributor {
+    login?: string | undefined;
+    id?: number | undefined;
+    node_id?: string | undefined;
+    avatar_url?: string | undefined;
+    gravatar_id?: string | null | undefined;
+    url?: string | undefined;
+    html_url?: string | undefined;
+    followers_url?: string | undefined;
+    following_url?: string | undefined;
+    gists_url?: string | undefined;
+    starred_url?: string | undefined;
+    subscriptions_url?: string | undefined;
+    organizations_url?: string | undefined;
+    repos_url?: string | undefined;
+    events_url?: string | undefined;
+    received_events_url?: string | undefined;
+    type?: string | undefined;
+    site_admin?: boolean | undefined;
+    contributions?: number | undefined;
+}
+
+export type RepoOwnerAndName = {
+    owner: string;
+    repo: string;
+};
+
+export interface Contributor {
+    login: string;
+}
+
+export class GitHubContributorFetcher {
+    public async fetchContributors(
+        token: string,
+        repositoryIdentifiers: string[]
+    ): Promise<Map<string, Contributor[]>> {
+        const ownerAndRepos = this.extractOwnerAndRepoNames(repositoryIdentifiers);
+
+        const gitHubContributorsByRepoName = await this.fetchGitHubContributorsByRepoName(
```

Comment:
Tiny thing, but the key for the returned map is really a repo identifier, not just the name part. So I would probably name the function and local `...ContributorsByRepo`, without "Name" at the end.